python/tk_framework_desktopserver/__init__.py

```diff
@@ -25,13 +26,24 @@
 _version_dir = "{}.{}".format(_py_version.major, _py_version.minor)
 
 if sgtk.util.is_macos():
-    sys.path.insert(0, os.path.join(binaries_path, _version_dir, "mac"))
+    binaries_os_path = os.path.join(binaries_path, _version_dir, "mac")
 elif sgtk.util.is_windows():
-    sys.path.insert(0, os.path.join(binaries_path, _version_dir, "win"))
+    binaries_os_path = os.path.join(binaries_path, _version_dir, "win")
 elif sgtk.util.is_linux():
-    sys.path.insert(0, os.path.join(binaries_path, _version_dir, "linux"))
+    binaries_os_path = os.path.join(binaries_path, _version_dir, "linux")
+
+if os.path.exists(binaries_os_path):
```

Comment:
This variable might not be defined when used here given the condition above.

Not super important but I understand that this PR is about Fault Tolerance to make sure this does not crash.

Also, when we don't find the path, should we "exit" or "raise" instead of continuing ignoring the error. Because it will just crash later isn't it?

Reply:
It should only be undefined if this is running in any other OS than macos, linux, and windows. But you're right a good practice is to define it at least with `None` value.

I wasn't sure about `raise` here on error. We're sure it will crash later. I like this idea, will implement it.